controllers/managedocs_controller.go

```diff
@@ -895,3 +903,117 @@ func (r *ManagedOCSReconciler) own(resource metav1.Object) error {
 	}
 	return nil
 }
+
+func (r *ManagedOCSReconciler) reconcileRookCephOperatorConfig() error {
+	rookConfigMap := &corev1.ConfigMap{}
+	rookConfigMap.Name = rookConfigMapName
+	rookConfigMap.Namespace = r.namespace
+
+	if err := r.get(rookConfigMap); err != nil {
+		// Because resource limits will not be set, failure to get the Rook ConfigMap results in failure to reconcile.
+		return fmt.Errorf("Failed to get Rook ConfigMap: %v", err)
+	}
+
+	if rookConfigMap.Data == nil {
+		rookConfigMap.Data = make(map[string]string)
+	}
+
+	rbdProvisionerLimits, err := yaml.Marshal([]struct {
+		Name     string                  `yaml:"name"`
+		Resource strResourceRequirements `yaml:"resource"`
+	}{
+		{
+			Name:     "csi-provisioner",
+			Resource: convertResourceRequirementsToString(utils.GetResourceConfig("csi-provisioner")),
```

Comment:
Since the `rook-ceph-operator-config configmap` expects the resources requirements to be in yaml format, I use yaml.marshal. However, `GetResourceConfig` returns `corev1.ResourceRequirements` which use Quantity objects to represent their values that do not marshal into the values that are needed. I added the helper `convertResourceRequirementsToString` so the the marshalling occurs properly.